src/VirtoCommerce.BackInStockModule.Data/BackgroundJobs/BackInStockNotificationJobService.cs

```diff
@@ -0,0 +1,158 @@
+            throw;
+        }
+    }
+
+    public async Task EnqueueBatchOfEmailNotificationsForProductIds(IList<string> inStockProductsIds)
+    {
+        foreach (var productId in inStockProductsIds)
+        {
+            try
+            {
+                var subscriptions = await FetchAllBackInStockSubscriptionsForProduct(productId);
+                foreach (var subscription in subscriptions)
+                {
+                    BackgroundJob.Enqueue(() =>
+                        SendBackInStockEmailNotificationAsync(subscription));
+                }
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, $"Failed to enqueue batch of notification jobs for product id: {productId}");
+                throw;
+            }
+        }
+    }
+
+    private async Task<IList<BackInStockSubscription>> FetchAllBackInStockSubscriptionsForProduct(string productId)
+    {
+        var allSubscriptions = new List<BackInStockSubscription>();
+        int totalCount, skip = 0;
+        do
+        {
+            var batchSize = await GetBatchSize();
+
+            var result = await backInStockSubscriptionSearchService.SearchAsync(
+                new BackInStockSubscriptionSearchCriteria
+                {
+                    ProductId = productId, IsActive = true, Skip = skip, Take = batchSize
+                });
+
+            allSubscriptions.AddRange(result.Results);
+            totalCount = result.TotalCount;
+            skip += batchSize;
+        }
+        while (allSubscriptions.Count < totalCount);
+
+        return allSubscriptions;
+    }
+
+    public async Task SendBackInStockEmailNotificationAsync(BackInStockSubscription subscription)
+    {
+        try
+        {
+            var notification = await PrepareNotification(subscription);
+            if (notification != null)
+            {
+                await notificationSender.ScheduleSendNotificationAsync(notification);
+            }
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, $"Failed to send {nameof(BackInStockEmailNotification)} notification");
+            throw;
+        }
+    }
+
+    private async Task<BackInStockEmailNotification> PrepareNotification(BackInStockSubscription subscription)
+    {
+        var notification = await notificationSearchService.GetNotificationAsync(
+                nameof(BackInStockEmailNotification),
+                new TenantIdentity(subscription.StoreId, nameof(Store)))
+            as BackInStockEmailNotification;
+
+        if (notification == null)
+        {
+            return null;
+        }
+
+        var customer = await memberResolver.ResolveMemberByIdAsync(subscription.UserId);
+        if (customer == null || !customer.Emails.Any())
+        {
+            return null;
+        }
+
+        var store = (await storeService.GetAsync(new List<string> { subscription.StoreId }))
+            .FirstOrDefault();
```

Comment:
Don't break lines like this.
Use `[subscription.StoreId]` instead of `new List<string> { subscription.StoreId }`
Use `storeService.GetByIdAsync(subscription.StoreId)`